server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -868,6 +868,8 @@ assert localNodeMayWinElection(getLastAcceptedState()) :
     // Package-private for testing
     ClusterState improveConfiguration(ClusterState clusterState) {
         assert Thread.holdsLock(mutex) : "Coordinator mutex not held";
+        assert validVotingConfigExclusionState(clusterState) : "Voting Config Exclusion in invalid state. " +
```

Comment:
We can deduce that it's this assertion that fails from the stack trace, so there's no need for this message, but it would be useful to see the cluster state when it fails:

```suggestion
        assert validVotingConfigExclusionState(clusterState) : clusterState;
```

Reply:
Done.